engine/exec/exec.go

```diff
@@ -7,18 +7,24 @@ package exec
 import (
 	"context"
 	"errors"
+	"fmt"
 	"io"
 	"os/exec"
+	"strings"
 
 	"github.com/drone/runner-go/pipeline/runtime"
 	"github.com/harness/lite-engine/engine/spec"
 )
 
-func Run(ctx context.Context, step *spec.Step, output io.Writer) (*runtime.State, error) {
+func Run(ctx context.Context, step *spec.Step, output io.Writer, isDrone bool) (*runtime.State, error) {
 	if len(step.Entrypoint) == 0 {
 		return nil, errors.New("step entrypoint cannot be empty")
 	}
 
+	// we shouldn't change this functionality for drone.
+	if !isDrone {
+		fmt.Fprintf(output, "Executing command:\n%s\n\n", strings.TrimSpace(strings.Join(step.Command, " ")))
```

Comment:
This should also be done on manager - the command formation for K8s/VMs is all done there. K8s also has this same logic of printing the commands initially before the output